src/js/jsx/tools/SuperselectVectorMaskOverlay.jsx

```diff
@@ -0,0 +1,346 @@
+                        .attr("fill", "#3691ff")
+                        .attr("mask", "url(#Mask)");
+                } else if (type === "rect") {
+                    svg.select("g#maskPreview").append("rect")
+                        .attr("x", bounds.left + boundsAB.left + offset)
+                        .attr("y", bounds.top + boundsAB.top + offset)
+                        .attr("width", bounds.width)
+                        .attr("height", bounds.height)
+                        .attr("fill", "#3691ff")
+                        .attr("opacity", 0.5);
+                }
+            }
+        },
+
+        /**
+         * Checks to see if the given position is under
+         * any always_propagate policies, where mouse events should go to PS
+         * and we shouldn't highlight layers
+         *
+         * @param {number} x
+         * @param {number} y
+         *
+         * @return {boolean} True if the pointer is under an always propagate policy
+         */
+        _positionUnderAlwaysPropagatePolicy: function (x, y) {
+            var policyStore = this.getFlux().store("policy"),
+                modifierStore = this.getFlux().store("modifier"),
+                modifiers = modifierStore.getState(),
+                modifierBits = keyUtil.modifiersToBits(modifiers),
+                pointerPolicies = policyStore.getMasterPointerPolicyList(),
+                underAlways = false;
+
+            // As soon as we find a policy that intersects and matches modifiers, we can return
+            pointerPolicies.some(function (policy) {
+                if (!policy.area || policy.modifiers !== modifierBits) {
+                    return false;
+                }
+
+                var area = {
+                        left: policy.area[0],
+                        top: policy.area[1],
+                        right: policy.area[0] + policy.area[2],
+                        bottom: policy.area[1] + policy.area[3]
+                    },
+                    intersects = area.left < x && area.right > x &&
+                        area.top < y && area.bottom > y;
+
+                if (intersects) {
+                    if (policy.action === UI.policyAction.PROPAGATE_TO_PHOTOSHOP ||
+                        policy.action === UI.policyAction.PROPAGATE_BY_ALPHA) {
+                        underAlways = true;
+                    } else {
+                        underAlways = false;
+                    }
+                }
+
+                return intersects;
+            }, this);
+
+            return underAlways;
+        },
+
+        /**
+         * Draws the clickable icons on the vector Mask HUD
+         *
+         * @private
+         */
+        _drawVectorMaskHUDObjects: function () {
+            var uiStore = this.getFlux().store("ui"),
+                cloakRect = uiStore.getCloakRect();
+            
+            var layerTree = this.state.document.layers;
+    
+            var fluxActions = this.getFlux().actions,
```

Comment:
you should put flux in a variable instead of calling `this.getFlux()` repeatedly.